source/extensions/filters/network/thrift_proxy/conn_manager.cc

```diff
@@ -28,6 +28,20 @@ Network::FilterStatus ConnectionManager::onData(Buffer::Instance& data, bool end
   request_buffer_.move(data);
   dispatch();
 
+  if (end_stream && stopped_) {
+    MessageMetadata& metadata = *(*rpcs_.begin())->metadata_;
```

Comment:
@zuercher is there any way that the iterator could be end()?

Reply:
No. The only way stopped gets set is if a an ActiveRpc object's filter chain requests it.

Reply:
should we leave an assertion behind?

Reply:
Sure. Can't hurt.